common/logger/logger.ts

```diff
@@ -0,0 +1,74 @@
+import { isCommandArg } from '../util/basic';
+import { configure, addLayout, getLogger as getlog4jsLogger, Logger as Log4jsLogger } from 'log4js';
+
+addLayout('json', function () {
+    return function (logEvent) {
+        const message = logEvent.data.shift();
+        const data = logEvent.data.shift();
+        return JSON.stringify({ ...logEvent, message, data });
+    };
+});
+
+let appenders = ['out'];
+if (process.env.LOG_FORMAT === 'json') {
+    appenders = ['outJson'];
+}
+
+configure({
+    appenders: {
+        out: { type: 'stdout', layout: { type: 'coloured' } },
+        outJson: { type: 'stdout', layout: { type: 'json' } },
+    },
+    categories: {
+        default: {
+            appenders,
+            level: isCommandArg('--debug') ? 'debug' : 'info',
+        },
+    },
+});
+
+export type LogData = { [key: string]: any };
+
+export class Logger {
+    constructor(private logger: Log4jsLogger) {}
+
+    getLoggerImpl(): any {
+        return this.logger;
+    }
+
+    trace(message: any, args: LogData = {}): void {
```

Comment:
Don't we want message to always be a string?

Reply:
Good catch!

Reply:
I'll have to fix quite a lot of logs now, but it needs to be done 😆

Reply:
We could also do multiple PRs

Reply:
Good Idea.
But in this case I would keep it in the same PR even tho it's going to be even more changes, because it's easier to rollback.